### PR TITLE
Fixed a problem with date calculations that caused a test to fail in a non-US time zone.

### LINKED ISSUE #60370: Date Picker keyboard navigation test fails when run in the Paris timezone
(cd packages/flutter; export TZ='Europe/Paris'; flutter test --enable-experiment=non-nullable test/material/date_picker_test.dart)

Will fail with:

```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure object was thrown running a test:
  Expected: DateTime:<2016-01-18 00:00:00.000>
  Actual: DateTime:<2016-01-17 00:00:00.000>

When the exception was thrown, this was the stack:
#4      main.<anonymous closure>.<anonymous closure>.<anonymous closure> (file:///Users/darrenaustin/dev/flutter/packages/flutter/test/material/date_picker_test.dart:1236:9)
<asynchronous suspension>
```

### PR DESCRIPTION
**Pull request by darrenaustin**
## Description

In order to navigate the date picker's grid with the keyboard we were computing the next/previous day by using `DateTime.add(Duration(days: -1))`. Unfortunately this uses daylight-saving changes in this computation, sometimes causing the computed day to be off by a day (instead of the time being 12:00 it is now 11:00 the previous day).

The PR addresses this by just calculating the days without using the `DateTime.add` function.

## Related Issues

Fixes: #60370

## Tests

This fixes the existing tests when running in non-US timezones.

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them?

- [x] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter/lib/src/material/pickers/date_utils.dart [hunk 1]
```diff
@@ -72,6 +72,12 @@ DateTime addMonthsToMonthDate(DateTime monthDate, int monthsToAdd) {
   return DateTime(monthDate.year, monthDate.month + monthsToAdd);
 }
 
+/// Returns a [DateTime] with the added number of days and truncates any day
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/pickers/date_utils.dart:75]: Not sure what the "and truncates any..." part of the doc is referring to.

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Comment by flutter-github-sync** [comment]
Google testing passed!

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Comment by flutter-github-sync** [comment]
Google testing passed!

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Comment by flutter-github-sync** [comment]
Google testing passed!

**Review by rami-a — APPROVED** [review]
LGTM

**Pull request merged**